scalalib/src/mill/scalalib/scalafmt/ScalafmtWorker.scala

```diff
@@ -0,0 +1,58 @@
+package mill.scalalib.scalafmt
+
+import ammonite.ops.{Path, exists}
+import mill._
+import mill.define.{Discover, ExternalModule, Worker}
+import mill.modules.Jvm
+import mill.util.Ctx
+
+import scala.collection.mutable
+
+object ScalafmtWorkerModule extends ExternalModule {
```

Comment:
It wasn't immediately clear that worker must live in a separate module to be a single one in a build. 
Also, I copy-pasted `extends ExternalModule` from scala and scalajs workers. Why should it be an `ExternalModule`, not just a `Module` ?
When I tried `extends Module` complilation failed with:

```
[error] /Users/rockjam/projects/mill/scalalib/src/mill/scalalib/scalafmt/ScalafmtWorker.scala:11:54: Modules, Targets and Commands can only be defined within a mill Module
[error] private[scalafmt] object ScalafmtWorker extends mill.Module {
```

Reply:
@rockjam being an `ExternalModule` allows a task `myTask` to be run via `mill.scalalib.scalafmt.ScalafmtWorkerModule/myTask`. Normal `Module` objects must be defined in the `build.sc` file and be run via their object selector from the root of `build.sc`. 

If you want to let people run ScalaFmt without needing to extend anything, you could make a task in the `ExternalModule` that lets people run e.g. `mill.scalalib.scalafmt.ScalafmtWorkerModule/fmt __.sources` to format the `sources` of all their `ScalaModule`s. See how the `mill.scalalib.PublishModule/publishAll` command works as an example.

As for being separate Mill submodules, we only needed to put things like `ScalaWorker`/`ScalaJSWorker` in there because we needed code that runs with some heavyweight dependencies on the classpath that might collide (e.g. different Scala.js versions): the dependencies are isolated within a classloader and interacted with via reflection. If you don't have any heavyweight dependencies, or you do but you can just interact with them directly via reflection/subprocesses, then there's no need to isolate them in a separate Mill build module